queue_services/entity-filer/tests/unit/filing_processors/test_annual_report.py

```diff
@@ -14,33 +14,49 @@
 """The Test Suites to ensure that the worker is operating correctly."""
 import copy
 import datetime
+import pytest
 import random
 from unittest.mock import patch
 
 from freezegun import freeze_time
-from legal_api.models import Business, Filing
+from legal_api.models import BatchProcessing, Business, Filing
 from registry_schemas.example_data import ANNUAL_REPORT
 
 # from entity_filer.filing_processors.filing_components import create_party, create_role
 from entity_filer.filing_meta import FilingMeta
 from entity_filer.worker import process_filing
 from tests.unit import (
+    create_batch,
+    create_batch_processing,
     create_business,
     create_filing,
 )
 
 
-def test_process_ar_filing(app, session):
+@pytest.mark.parametrize('test_name,flag_on,in_dissolution,eligibility,legal_type,', [
+    ('AR successfully', True, False, False, 'CP'),
+    ('Not withdrawn from the dissolution process', True, True, True, 'CP'),
+    ('Withdrawn from the dissolution process', True, True, False, 'BC'),
+    ('AR successfully when flag is off', False, True, False, 'CP')
+])
+def test_process_ar_filing_involuntary_dissolution(app, session, test_name, in_dissolution, eligibility, legal_type, flag_on):
     """Assert that an AR filling can be applied to the model correctly."""
-    from entity_filer.worker import APP_CONFIG
     from entity_filer.filing_processors import annual_report
     # vars
     payment_id = str(random.SystemRandom().getrandbits(0x58))
     identifier = 'CP1234567'
 
-    # setup
-    business = create_business(identifier, 'CP')
-    business_id = business.id
+    business = create_business(identifier, legal_type)
+    # create the batch and batch_processing.
+    batch_status = 'PROCESSING'
+    if not in_dissolution:
+        batch_status = 'COMPLETED'
+    batch = create_batch(status=batch_status)
+    batch_processing = create_batch_processing(business=business, batch_id=batch.id, status=batch_status)
+
+    business.no_dissolution = False if eligibility else True
```

Comment:
How you are using this(` business.no_dissolution`) is bypassing testing what we want to test at least in some scenarios from what I can see.

We need to be able to confirm that the eligibility checks are happening based on the AR related date fields and not bypass the checks because we have used `business.no_dissolution` flag.